nemo/collections/asr/modules/audio_modules.py

```diff
@@ -1668,16 +1668,18 @@ def estimate_filter(self, Q: torch.Tensor, R: torch.Tensor) -> torch.Tensor:
             Q = Q + torch.diag_embed(diag_reg.unsqueeze(-1) * torch.ones(Q.shape[-1], device=Q.device))
 
         # Solve for the filter
+        fail = False
         try:
             # Use Cholesky decomposition
             QL = torch.linalg.cholesky(Q)
             # Forward and backward substitution
             G = torch.linalg.solve_triangular(QL, R, upper=False)
             G = torch.linalg.solve_triangular(QL.conj().transpose(-2, -1), G, upper=True)
         except torch.linalg.LinAlgError as e:
-            # Fallback to linsolve
-            logging.warning('Cholesky failed with exception: %s', e)
-            logging.info('Fallback to linsolve')
+            fail = True
```

Comment:
@tango4j this caused memory leakage in GSS processing. 
see https://pytorch.org/docs/stable/notes/faq.html#my-out-of-memory-exception-handler-can-t-allocate-memory